File: mliajava/src/main/java/classifier/kNN/kNNTest.java
package classifier.kNN;

import org.nd4j.linalg.api.ndarray.INDArray;
import org.nd4j.linalg.factory.Nd4j;

import static classifier.kNN.kNN.classify0;
import static java.lang.System.out;

public class kNNTest {
    public static Object[] createDataSet() {
        INDArray dataSet = Nd4j.create(new double[]{1.0f, 1.1f, 1.0f, 1.0f, 0.0f, 0.0f, 0.0f, 1.1f}, new int[]{4, 2});
        String[] labels = new String[]{"A", "A", "B", "B"};
        return new Object[]{dataSet, labels};
    }

    public static void main(String[] args) {
        Object[] obs = createDataSet();
        INDArray inX = Nd4j.create(new double[]{1.0f, 1.1f}, new int[]{1, 2});
        out.printf("classify: %s\n", classify0(inX, (INDArray) obs[0], (String[]) obs[1], 1));
    }

}

File: mliascala/src/main/scala/mlia_breeze/classifier/kNN/kNNTest.scala
package mlia_breeze.classifier.kNN

import breeze.linalg.Matrix
import breeze.linalg._
import breeze.stats.distributions._
import breeze.numerics._

object kNNTest extends App{
  def createDataSet(): Tuple2[Matrix[Double], Vector[String]] = {
    val dataSet = DenseMatrix.zeros[Double](4, 2)
    dataSet(0, ::).:=(DenseVector[Double](1.0, 1.1).t)
    dataSet(1, ::).:=(DenseVector[Double](1.0, 1.0).t)
    dataSet(2, ::).:=(DenseVector[Double](0, 0).t)
    dataSet(3, ::).:=(DenseVector[Double](0, 1.1).t)
    val labels = DenseVector[String]("A", "A", "B", "B")
    return Tuple2(dataSet, labels)
  }
  
  override def main(args: Array[String]): Unit = {
    val (group, labels) = kNNTest.createDataSet()
    val inX = DenseVector[Double](0.0, 1.1)

    printf("classify: %s\n", kNN.classify0(inX, group, labels, 1))
  }
}